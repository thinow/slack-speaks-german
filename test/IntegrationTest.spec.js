const nock = require('nock')
const {handlers} = require('../index')
const RequestBodyCaptor = require('./common/RequestBodyCaptor')

describe('when triggering the handler', () => {
    it('should have sent a message to the webhook', async () => {
        // given
        const requestBodyCaptor = new RequestBodyCaptor()
        nock('https://mocked-slack-host/')
            .post('/webhook')
            .reply(requestBodyCaptor.reply(200, 'ok'))

        // when
        await handlers({
            webhook: 'https://mocked-slack-host/webhook',
            wordsFolder: './test/resources/folder-single-word'
        })

        // then
        expect(requestBodyCaptor.getValue()).toMatchSnapshot()
    })
})
