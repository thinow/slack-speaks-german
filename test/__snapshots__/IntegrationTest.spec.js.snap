// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when triggering the handler should have sent a message to the webhook 1`] = `
Object {
  "attachments": Array [
    Object {
      "blocks": Array [
        Object {
          "elements": Array [
            Object {
              "text": "Wort des Tages",
              "type": "mrkdwn",
            },
          ],
          "type": "context",
        },
        Object {
          "text": Object {
            "text": "<https://www.duden.de/rechtschreibung/einzig|*einzig*>   __
only, unique",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        Object {
          "type": "divider",
        },
        Object {
          "text": Object {
            "text": "*Wir waren die einzigen GÃ¤ste.*
We were the only guests.",
            "type": "mrkdwn",
          },
          "type": "section",
        },
        Object {
          "text": Object {
            "text": "*Das war das einzige Mal, dass ich ihn gesehen habe.*
That was the only time I saw him.",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
      "color": "#3583f3",
    },
  ],
}
`;
